### PR TITLE
+ Test with Fable

### PR DESCRIPTION
**Pull request by wallymathieu**
- async test case in order to be able to set a timeout for a test that can be awaited in an outer test in node
- test project that can be use to test running Fuchu with Fable
- [x] make sure that the test project runs the tests and reports success/failure
- keep compatibility with existing code

### CODE REVIEW

**Comment by wallymathieu** [comment]
For some reason appveyor and github actions doesn't trigger

**Comment by wallymathieu** [comment]
I've gotten so far that it compiles. The odd thing right now is that while starting it with node, it seems to exit before running the tests. Any ideas @alfonsogarciacaro ?

File: package.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "private": true,
+  "scripts": {
+    "pretest": "dotnet fable Fuchu.FableTests --outDir bin",
+    "test": "node --eval \"require('esm')(module)('./bin/Main.js')\" ./bin/Main.js ./bin/Main.js ./bin "
```
Comment (alfonsogarciacaro) [inline package.json:5]: Not sure if that's the reason, but I usually run the tests like this: `npx mocha ./bin -r esm`

**Comment by alfonsogarciacaro** [comment]
@wallymathieu It may be that the test runner doesn't find any test. Can you try running the tests with `npx mocha ./bin -r esm`. Make sure the files in that folder contain tests in Mocha format: https://mochajs.org/#getting-started

**Comment by wallymathieu** [comment]
The thing is that Fuchu is a test runner written in F#. Since it's less than 1000 lines of code I'm thinking that it would be neat to be able to use Fuchu instead of Mocha/Expecto and instead use a single runner for both .net and Fable (given some adjustment of Fuchu).

**Comment by alfonsogarciacaro** [comment]
Ah, ok! Yes, an F# test runner for Fable would be nice. I will try to clone and reproduce.

**Comment by wallymathieu** [comment]
Right now I've found that I'm probably doing something wrong related to async things. The current code hangs in a setTimeout while loop.

**Comment by wallymathieu** [comment]
I think I did something silly indeed, the Async.RunSynchronously polyfill didn't work as intended.

File: package.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
+  "private": true,
+  "scripts": {
+    "pretest": "dotnet fable Fuchu.FableTests --outDir bin",
+    "test": "node --eval \"require('esm')(module)('./bin/Main.js')\" ./bin/Main.js ./bin/Main.js ./bin /d"
```
Comment (alfonsogarciacaro) [inline package.json:5]: Sorry @wallymathieu, forgot to tell there's an option in Fable `--runScript` that basically does this. I think something like this should work:

```
"test": "dotnet fable Fuchu.FableTests --outDir bin --runScript ./bin /d"
```

**Pull request merged**